lib/Settings/Settings.h

```diff
@@ -36,6 +36,13 @@
 #define RECEIVER_PIN 12  // avoid 0, 2, 15, 16
 #define TRANSMITTER_PIN 4
 
+#ifndef FIRMWARE_VERSION
+#define FIRMWARE_VERSION unknown
+#endif
+
+#define X_QUOTE(x) #x
+#define QUOTE(x) X_QUOTE(x)
```

Comment:
This looks very complicated. Why not simply use `#FIRMWARE_VERSION` instead of `QUOTE(FIRMWARE_VERSION)`? But even better would be:
```cpp
const char[] firmwareVersion = #FIRMWARE_VERSION
```
or similar (String or PROGMEM) and then simply use this variable in `MqttClient.cpp` and `ConfigWebServer.cpp`.

Reply:
This is because we give the version as a commandline argument to the compiler and this needs to get quotet to be a string literal. This snipped is what everyone else seems to do.

Reply:
And you cannot use macros in a String literal. So 
```
const char[] firmwareVersion = "#FIRMWARE_VERSION"
```
would also not work.